src/DotVVM.Framework/Diagnostics/CompilationPageApiPresenter.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Threading.Tasks;
+using DotVVM.Framework.Compilation;
+using DotVVM.Framework.Configuration;
+using DotVVM.Framework.Hosting;
+using Newtonsoft.Json;
+
+namespace DotVVM.Framework.Diagnostics
+{
+    internal class CompilationPageApiPresenter : IDotvvmPresenter
+    {
+        private readonly IDotvvmViewCompilationService compilationService;
+
+        public CompilationPageApiPresenter(IDotvvmViewCompilationService compilationService)
+        {
+            this.compilationService = compilationService;
+        }
+
+        public async Task ProcessRequest(IDotvvmRequestContext context)
+        {
+            var response = context.HttpContext.Response;
+            var isAuthorized = await context.Configuration.Diagnostics.CompilationPage.AuthorizationPredicate(context);
+            if (!isAuthorized)
+            {
+                response.StatusCode = 403;
+                return;
+            }
+
+            var result = await compilationService.CompileAll(false);
```

Comment:
```suggestion
            var result = await compilationService.CompileAll(buildInParallel: false);
```

And actually, why not build in parallel by default?

Reply:
Not sure. I don't see a reason why not. This was taken directly from the status page repo.